tests/Configuration/Connections/MasterSlaveConnectionTest.php

```diff
@@ -0,0 +1,259 @@
+
+        // Case #3. Simple valid configuration with pgqsql base settings.
+        $expectedConfigPgsql                         = $expectedConfig;
+        $expectedConfigPgsql['driver']               = 'pgsql';
+        $expectedConfigPgsql['master']['user']       = 'homestead1';
+        $expectedConfigPgsql['master']['sslmode']    = 'sslmode';
+        $expectedConfigPgsql['slaves'][0]['sslmode'] = 'sslmode';
+        $expectedConfigPgsql['slaves'][1]['sslmode'] = 'sslmode';
+
+        $resolvedBaseSettings = [
+            'driver'      => 'pgsql',
+            'host'        => 'localhost',
+            'dbname'      => 'test',
+            'user'        => 'homestead',
+            'password'    => 'secret',
+            'port'        => 'port',
+            'sslmode'     => 'sslmode',
+        ];
+        $out[] = [$resolvedBaseSettings, $dummyInputConfig, $expectedConfigPgsql];
+
+        // Case #4. Simple valid configuration with sqlite base settings.
+        $inputConfigSqlite = $dummyInputConfig;
+        unset($inputConfigSqlite['read'][0]['database']);
+        unset($inputConfigSqlite['read'][1]['database']);
+        unset($inputConfigSqlite['write']['database']);
+
+        $expectedConfigSqlite = [
+            'wrapperClass' => MasterSlaveDoctrineWrapper::class,
+            'driver'       => 'pdo_sqlite',
+            'slaves'       => [
+                [
+                    'user'     => 'homestead',
+                    'password' => 'secret',
+                    'port'     => 3308,
+                    'path'     => ':memory',
+                    'memory'   => true,
+                ],
+                [
+                    'host'     => 'localhost2',
+                    'user'     => 'homestead',
+                    'password' => 'secret',
+                    'port'     => 3309,
+                    'path'     => ':memory',
+                    'memory'   => true,
+                ]
+            ],
+            'master' => [
+                'user'     => 'homestead1',
+                'password' => 'secret1',
+                'port'     => 3307,
+                'memory'   => true,
+                'path'     => ':memory',
+            ],
+        ];
+
+        $resolvedBaseSettings = [
+            'driver'   => 'pdo_sqlite',
+            'path'     => ':memory',
+            'user'     => 'homestead',
+            'password' => 'secret',
+            'memory'   => true
+        ];
+        $out[] = [$resolvedBaseSettings, $inputConfigSqlite, $expectedConfigSqlite];
+
+        return $out;
+    }
+
+    /**
+     * Check if master slave connection manages configuration well.
+     *
+     * @param array $resolvedBaseSettings
+     * @param array $settings
+     * @param $expectedOutput
+     *
+     * @dataProvider getMasterSlaveConnectionData
+     */
+    public function testMasterSlaveConnection(array $resolvedBaseSettings, array $settings, array $expectedOutput)
+    {
+        $this->assertEquals(
```

Comment:
I like how you used dataProvider to reuse this case multiple times, but I find it a bit hard to maintain in this current form.
I wouldn't know where to modify each expectation for each setting.

Maybe refactor each case into its own method, then do something like:

```
public function getMasterSlaveConnectionData()
{
    return [
        $this->dummyBaseSettings(), $this->dummyConfig(), $this->dummyExpectations(),
        $this->oracleBaseSettings(), $this->oracleConfig(), $this->oracleExpectations(),
}
```

Or something along those lines?